sdks/java/core/src/main/java/com/google/cloud/dataflow/sdk/io/BigQueryIO.java

```diff
@@ -975,18 +990,20 @@ public PDone apply(PCollection<TableRow> input) {
         String tempLocation = options.getTempLocation();
         checkArgument(!Strings.isNullOrEmpty(tempLocation),
             "BigQueryIO.Write needs a GCS temp location to store temp files.");
-        try {
-          GcsPath.fromUri(tempLocation);
-        } catch (IllegalArgumentException e) {
-          throw new IllegalArgumentException(String.format(
-              "BigQuery temp location expected a valid 'gs://' path, but was given '%s'",
-              tempLocation), e);
+        if (testBigQueryServices == null) {
+          try {
+            GcsPath.fromUri(tempLocation);
+          } catch (IllegalArgumentException e) {
+            throw new IllegalArgumentException(String.format(
+                "BigQuery temp location expected a valid 'gs://' path, but was given '%s'",
+                tempLocation), e);
+          }
         }
-
         String jobIdToken = UUID.randomUUID().toString();
         String tempFilePrefix = tempLocation + "/BigQuerySinkTemp/" + jobIdToken;
-        String jsonTable = toJsonString(table);
+        String jsonTable = toJsonString(checkNotNull(table, "table"));
         String jsonSchema = toJsonString(schema);
```

Comment:
presumably we will not already have this as a JSON string once the above fix is made.